Added OnePlayer events (#7415)
MicrosoftDocs/OfficeDocs-SkypeForBusiness: Teams/policy-control-diagnostic-data-mobile.md

| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|
| Trace_message | Contains error string and details about the reasons due to which a failure might have happened |

## OnePlayer Events

> ### OnePlayer User Action Events

Reviewer: ```suggestion
### OnePlayer user action events
```

> - **PlayerPlay** - Confirms if the user taps on the play button in the OnePlayer view
> - **PlayerPause** - Confirms if the user taps on the pause button in the OnePlayer view
> - **PlayerSeek** - Confirms if the user seeks the video either using seek bar or forward/backward buttons in the OnePlayer view [iOS Only]
> - **VideoPlayerSeekForward** - Confirms if the user seeks the video either using seek bar or forward buttons in the OnePlayer view [Android Only]
> - **VideoPlayerSeekBackward** - Confirms if the user seeks the video either using seek bar or backward buttons in the OnePlayer view [Android Only]
> - **ChangePlaybackSpeed** - Confirms if the user has selected a new playback speed
> - **changePlaybackQuality** - Confirms if the user has selected a new video quality for playback
> - **ShareVideo** - Confirms if the user has tapped on share icon
> - **PlayerClose** - Confirms if the user has tapped on the close icon
> - **VideoCaptionsOn** - Confirms if the user has switched on the caption
> - **VideoCaptionsOff** - Confirms if the user has switched off the caption
> - **ChangePlayerOrientation** - Confirms if the user has changed orientation of the device
> - **OpenPlayerSettingsMenu** - Confirms if the user has opened settings menu
> - **OpenPlaybackSpeedMenu** - Confirms if the user has opened playback speed menu
> - **PlayerAction** - Custom action provided by host app

Reviewer: ```suggestion
- **PlayerPlay** - Confirms if the user taps on the play button in the OnePlayer view.
- **PlayerPause** - Confirms if the user taps on the pause button in the OnePlayer view.
- **PlayerSeek** - Confirms if the user seeks the video either using the seek bar or forward/backward buttons in the OnePlayer view (iOS only).
- **VideoPlayerSeekForward** - Confirms if the user seeks the video either using the seek bar or forward buttons in the OnePlayer view (Android only).
- **VideoPlayerSeekBackward** - Confirms if the user seeks the video either using the seek bar or backward buttons in the OnePlayer view (Android only).
- **ChangePlaybackSpeed** - Confirms if the user has selected a new playback speed.
- **changePlaybackQuality** - Confirms if the user has selected a new video quality for playback.
- **ShareVideo** - Confirms if the user has tapped on the share icon.
- **PlayerClose** - Confirms if the user has tapped on the close icon.
- **VideoCaptionsOn** - Confirms if the user has switched on the captions.
- **VideoCaptionsOff** - Confirms if the user has switched off the captions.
- **ChangePlayerOrientation** - Confirms if the user has changed the orientation of the device.
- **OpenPlayerSettingsMenu** - Confirms if the user has opened the settings menu.
- **OpenPlaybackSpeedMenu** - Confirms if the user has opened the playback speed menu.
- **PlayerAction** - Custom action provided by the host app.
```

> ### OnePlayer Playback Events
> - **PlayerHeartbeat** - This is recurring event sent to log the current status of player and playback

Reviewer: ```suggestion
### OnePlayer playback events
- **PlayerHeartbeat** - This is a recurring event that sends the current status of the player and playback to a log.
```

> ### a) Properties sent with all OnePlayer events
> ##### a.1 Standard Properties

Reviewer: ```suggestion
### 1. Properties sent with all OnePlayer events
##### 1.1 Standard properties
```

| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|

> | eventType | type of event (AppLogic, ErrorAlert, Performance, UserAction) |
> | accountType   | type of user account (business etc.) |
> | component     | OnePlayer |
> | language      | locale/ language of the app |
> | platform      | platform of OnePlayer(iOS / Android) |
> | tenantId      | tenant id |
> | version       | version of the OnePlayer being userd |
> | aadUserId     | user id of the user |                                

Reviewer: ```suggestion
| eventType | Type of event (AppLogic, ErrorAlert, Performance, UserAction) |
| accountType   | Type of user account (for example, business) |
| component     | OnePlayer |
| language      | Locale/language of the app |
| platform      | Platform of OnePlayer (iOS/Android) |
| tenantId      | Tenant ID|
| version       | Version of the OnePlayer being used |
| aadUserId     | User ID |                                
```

> ##### a.2 Player Properties

Reviewer: ```suggestion
##### 1.2 Player Properties
```

| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|

> | engineName    | Underlying player name (AVFoundation for iOS / ExoPlayer for Android) |
> | engineVersion | Operating System Version |
> | loadMode      | load mode of the player |
> | playbackSessionId | session id for playback |

Reviewer: ```suggestion
| engineName    | Underlying player name (AVFoundation for iOS/ExoPlayer for Android) |
| engineVersion | Operating System version |
| loadMode      | Load mode of the player |
| playbackSessionId | Session ID for playback |
```

> ##### a.3 Host Properties 

Reviewer: ```suggestion
##### 1.3 Host properties 
```

| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|

> | hostIntegrationType | host integration type(Package, OneUp etc.) |
> | hostPlatform  | platform for host app |
> | hostProperties| host properties if any [iOS only] |
> | hostApp       | name of the host app |
> | hostVersion   | version of the host app |

Reviewer: ```suggestion
| hostIntegrationType | Host integration type (for example, Package, OneUp) |
| hostPlatform  | Platform for host app |
| hostProperties| Host properties if any (iOS only) |
| hostApp       | Name of the host app |
| hostVersion   | Version of the host app |
```

> ##### a.4 Experimentation Properties

Reviewer: ```suggestion
##### 1.4 Experimentation properties
```

| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|

> | ring          | ring to which user belongs |
> | hostSettings  | attributes set by host app (moreOptionsEnabled, shareFeatureEnabled, playbackQualityFeatureEnabled, playbackSpeedFeatureEnabled) |
> | flightFilters | description |
> | flightsOverridden | bool for flights overridden or not |

Reviewer: ```suggestion
| ring          | Ring to which the user belongs |
| hostSettings  | Attributes set by the host app (moreOptionsEnabled, shareFeatureEnabled, playbackQualityFeatureEnabled, playbackSpeedFeatureEnabled) |
| flightFilters | Description |
| flightsOverridden | Bool for flights overridden or not |
```

> ##### a.5 Service Properties
> | Property name | Description                                                                                    |
> |---------------|------------------------------------------------------------------------------------------------|
> | contentType   | type of content being served |
> | environment   | environment name  |
> | mediaService  | which media service is being used (SPO, ODB, ODC, IC3-AMS, Unknown) |
> | mediaType     | type of the media being played  |
> | playbackTech  | playback tech of the media  |
> | correlationId | correlation id for the media if any |
> 

Reviewer: ```suggestion
##### 1.5 Service properties
| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|
| contentType   | Type of content being served |
| environment   | Environment name  |
| mediaService  | Which media service is being used (SPO, ODB, ODC, IC3-AMS, Unknown) |
| mediaType     | Type of the media being played  |
| playbackTech  | Playback tech of the media  |
| correlationId | Correlation ID for the media, if any |
```

> ### b) Properties sent with all OnePlayer User Action Events [iOS only]
> | Property name | Description                                                                                    |
> |---------------|------------------------------------------------------------------------------------------------|
> | actionType    | type of action being performed like tap, drag, flick etc. |
> | isIntentional | boolean value if the action is intentional or not |
> 
> #### b.1 Properties sent with changePlaybackQuality Event
> | Property name | Description                                                                                    |
> |---------------|------------------------------------------------------------------------------------------------|
> | currentPlaybackQuality | current playback quality |
> 

Reviewer: ```suggestion
### 2. Properties sent with all OnePlayer user action events (iOS only)
| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|
| actionType    | Type of action being performed, such as tap, drag, and flick |
| isIntentional | Boolean value if the action is intentional or not |

#### b.1 Properties sent with changePlaybackQuality Event
| Property name | Description                                                                                    |
|---------------|------------------------------------------------------------------------------------------------|
| currentPlaybackQuality | current playback quality |

```

> #### b.2 Properties sent with ChangePlaybackSpeed Event
> | Property name | Description |
> |---------------|------------------------------------------------------------------------------------------------|
> | previousPlaybackRate  | previous playback rate of the video [iOS only] |
> | currentPlaybackRate   | current playback rate of the video |
> 
> #### b.3 Properties sent with PlayerSeek Event [iOS only]
> | Property name | Description |
> |---------------|------------------------------------------------------------------------------------------------|
> | seekSource    | source of seek (seekbar, forwardButton, backwardButton) |
> | seekValue     | seek position |

Reviewer: ```suggestion
#### 2.2 Properties sent with ChangePlaybackSpeed Event
| Property name | Description |
|---------------|------------------------------------------------------------------------------------------------|
| previousPlaybackRate  | Previous playback rate of the video (iOS only) |
| currentPlaybackRate   | Current playback rate of the video |

#### 2.3 Properties sent with PlayerSeek Event [iOS only]
| Property name | Description |
|---------------|------------------------------------------------------------------------------------------------|
| seekSource    | Source of seek (seekbar, forwardButton, backwardButton) |
| seekValue     | Seek position |
```

> ### c) Properties sent only with Heartbeat Event

Reviewer: ```suggestion
### 3. Properties sent only with Heartbeat Event
```

| Property name | Description |
|---------------|------------------------------------------------------------------------------------------------|

> | mediaCurrentTime | current playback time of the media [iOS only]|
> | isLoaded | is media loaded |
> | loadTimeMs | load time taken in milliseconds |
> | numberOfStalls | number of stalls during playback [iOS only] |
> | bufferingCount | number of stalls during playback [Android only] |
> | observedBitrate | observed bit rate during playback [iOS only] |
> | avgBitrateBitsPerSecond | observed bit rate during playback [Android only] |
> | playedSeconds | played seconds till the event |
> | rebufferingSeconds | rebuffering seconds during playback |
> | timeSinceSourceSetMs | item since source was set ms |
> | triggerType | trigger type (buffering, error, errorLog, canPlayThrough, intervalHeartbeat, sourceset, unload) |
> | errorId | error id for the error if any |
> | errorCorrelationId | error correlation id for the error if any |
> | errorLog | error log for the error if any |
> | errorType | error type for the error if any |
> | errorMessage | error message for the error if any |
> | errorStack | extended error info for the error if any |
> | metaUrl | meta url for media |
> | odspDocId | odsp doc id for media |
> | siteId | site id for media |
> | teamsCallId | teams call id for media if any |

Reviewer: ```suggestion
| mediaCurrentTime | Current playback time of the media (iOS only)|
| isLoaded | Is media loaded |
| loadTimeMs | Load time taken in milliseconds |
| numberOfStalls | Number of stalls during playback (iOS only) |
| bufferingCount | Number of stalls during playback (Android only) |
| observedBitrate | Observed bit rate during playback (iOS only) |
| avgBitrateBitsPerSecond | Observed bit rate during playback (Android only) |
| playedSeconds | Played seconds till the event |
| rebufferingSeconds | Rebuffering seconds during playback |
| timeSinceSourceSetMs | Time since source was set (ms) |
| triggerType | Trigger type (buffering, error, errorLog, canPlayThrough, intervalHeartbeat, sourceset, unload) |
| errorId | Error ID for the error, if any |
| errorCorrelationId | Error correlation ID for the error, if any |
| errorLog | Error log for the error, if any |
| errorType | Error type for the error, if any |
| errorMessage | Error message for the error, if any |
| errorStack | Extended error info for the error, if any |
| metaUrl | Meta URL for media |
| odspDocId | ODSP doc ID for media |
| siteId | Site ID for media |
| teamsCallId | Teams call ID for media, if any |
```
